### ISSUE TITLE
More granular exporting

### ISSUE THREAD

**Issue by shaffan33** [dfinke/ImportExcel#70]
So I am working with the export-excel function within powershell and it seems to do most of what I am looking for, but I was wondering if there was a way to format the output more granularly?  For example, if I wanted to get the status of a specific service and put it in Column C, Row 12. Or if I then wanted to add headers, or text in a column. Would this be possible?  Ideally, I would like to use this to aggregate a bunch of data and put it in one excel sheet, for a FQC process of a server build.

**Comment by dfinke** [comment]
There was some work done by another community member, and not completed, around passing the Excel "object" back before it was saved so you could manipulate all the exposed properties of the workbook and worksheets and then save it.

**Comment by shaffan33** [comment]
Thanks.  Do you know where I could find that?

**Comment by dfinke** [comment]
This might help
https://github.com/dfinke/ImportExcel/issues/53#issuecomment-181937909

**Comment by shaffan33** [comment]
I think this would actually be all that I need:
https://github.com/dfinke/ImportExcel/blob/master/Examples/Tables/MultipleTables.ps1

How do I actually get the –startrow parameter though?  When I run it, I get the following:
“Export-Excel : A parameter cannot be found that matches parameter name
'startrow'.”

Andrew Shaffer
Enterprise Technology Services
617.664.5004

**Comment by dfinke** [comment]
Hi Andrew, 

It is here: 

https://github.com/dfinke/ImportExcel/blob/master/Export-Excel.ps1#l53

Hope this helps
Doug

**Comment by shaffan33** [comment]
Hmm- using that module I’m now getting the following:
Export-Excel : Unable to find type [OfficeOpenXml.Style.ExcelFillStyle].
At line:1 char:1
+ Export-Excel ESXtest.xlsx -title "ESX Host Name: TEST Ran on $(Get-Da ...
+ ~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (OfficeOpenXml.Style.ExcelFillStyle:TypeName) [], RuntimeException
    + FullyQualifiedErrorId : TypeNotFound

**Comment by dfinke** [comment]
Did you download this from GitHub or the Gallery?

**Comment by shaffan33** [comment]
Github.  I got it working now- thanks for all the help!  Unfortunately, I thought it wouldn’t overwrite text that was already present.  So I thought running something like:

Gsv | export-excel test.xlsx –startrow 1
Gsv | export-excel test.xlsx –startrow 300

Would produce two sets of GSV, instead of just one set starting at row 300.  Oh well, exporting data to different worksheets will have to do for now.

**Comment by dfinke** [comment]
That works for me. I get two gsv results, the second starting at row 300

**Comment by shaffan33** [comment]
That’s strange.  So what are the exact modules you are importing?  I have to install the default importexcel module in powershell and then the export-excel module you mentioned previously for it to work for me.  Once those are imported, I can run the command with the startrow parameter, but it just overwrites everything every time it is run.

**Comment by dfinke** [comment]
I'm running PS v5, so module autoloading is in play. Import-Module ImportExcel should do the trick if you have setup the module correctly. You may have issues if you're running v2 (maybe if v3).

Maybe clean out all the ImportExcel modules you have and then reinstall. If you're running PS v5, best to get it from the gallery.

**Comment by shaffan33** [comment]
I am using PS Version 5 as well.  I start a new powershell session and run the following:
PS C:\> import-module importexcel
This allows me to export to excel, but when I run the command with the startrow command, I get the below:
PS C:\> gsv | Export-Excel test34.xlsx -startrow 7
Export-Excel : A parameter cannot be found that matches parameter name 'startrow'.
At line:1 char:32
+ gsv | Export-Excel test34.xlsx -startrow 7
+                                ~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Export-Excel], ParameterBindingException
    + FullyQualifiedErrorId : NamedParameterNotFound,Export-Excel

When I then import the module you sent before (https://github.com/dfinke/ImportExcel/blob/master/Export-Excel.ps1#l53) I no longer get an error, but it does not append to the document, it simply overwrites whatever is already there.

**Comment by dfinke** [comment]
I'd remove the ImportExcel module. Close and open the console. Install the module from the gallery. Close and open the console. Then try this again. 

If if fails, need more details about your env.

**Comment by shaffan33** [comment]
So when I don’t use the importexcel module, and only use the module you sent earlier in this thread, I get the below:
PS C:\temp> gsv | export-excel c:\temp\test.xlsx
export-excel : Unable to find type [OfficeOpenXml.Style.ExcelFillStyle].
At line:1 char:7
+ gsv | export-excel c:\temp\test.xlsx
+       ~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (OfficeOpenXml.Style.ExcelFillStyle:TypeName) [], RuntimeException
    + FullyQualifiedErrorId : TypeNotFound

**Comment by dfinke** [comment]
see the previous comment

**Comment by shaffan33** [comment]
I got it from the gallery.  I realized I did not have the latest version, so I updated it:
PS C:\temp> Install-Module -name ImportExcel -requiredversion 1.98
and I can export, but now the startrow command does not work again:
PS C:\temp> gsv | export-excel c:\temp\test.xlsx -startrow 1
Export-Excel : A parameter cannot be found that matches parameter name 'startrow'.

I can then load the module you posted earlier, but again, it will just overwrite whatever was put in the spreadsheet before.
